go-ethereum-guides/write-basic-quiz-dapp-in-go/write-basic-quiz-dapp-in-go.md

```diff
@@ -0,0 +1,1069 @@
+check out this [Ethereum 101 guide](https://beta.kauri.io/article/67a81d8746ee4b49ba19447e8e2a983e/v2)
+instead.
+
+To start developing our DApp using an endpoint provided by Infura.io:
+
+1. Go to [Infura.io](https://infura.io) and sign up for an account.
+2. Once you've signed up, go to your Dashboard and click on **CREATE NEW PROJECT**.
+3. Enter a name for your project, 
+and click **CREATE** to set up a new project that we'll be using for our DApp.
+
+Your newly created project should look like this:
+
+![New project on Infura.io](./assets/infura-new-project.jpg)
+
+We'll come back to this later, when we've deployed our smart contract.
+
+For now, we need the URL of our project's Ethereum API gateway endpoint. 
+Select "RINKEBY" from the **ENDPOINT** dropdown menu, 
+and take note of the URL that appears underneath it. It should look like this: 
+
+```
+https://rinkeby.infura.io/v3/<PROJECT_ID>
+```
+
+**IMPORTANT!:** Make sure that the endpoint you're using in your Go code
+points to the Rinkeby testnet. If you use an endpoint pointing to the
+Ethereum mainnet, you'll be spending real Ether to test your application.
+
+Creata a file in your project folder named `.env`.
+**Do not** commit this file to Git 
+or any other version control system (VCS) you might be using.
+Edit the `.env` file and enter your project's Ethereum API gateway endpoint like so:
+
+```bash
+GATEWAY="https://rinkeby.infura.io/v3/<PROJECT_ID>"
+```
+
+Save the file. We'll be loading this into our Go application later.
+
+**NOTE:** Using a third-party provider to connect to the Ethereum network means 
+that you're trusting it with all transactions and any Ether that you send through it. 
+If you choose to not use a third-party provider, 
+you'll have to either run and host your own Ethereum API gateway,
+or rely on your users to connect to their own Ethereum nodes.
+
+### Set up an Ethereum account
+
+We'll need an Ethereum account to deploy our smart contract.
+If you don't have an existing Ethereum account, 
+you can create one using the Geth tool that comes with the
+Go Ethereum SDK.
+
+To create a new Ethereum account, run in the terminal:
+
+```bash
+geth --datadir . account new
+```
+
+and follow the on-screen instructions.
+
+This creates a keystore folder in the current directory. 
+In it, you'll find a keystore file for your new account.
+It should look like this:
+`UTC--<timestamp>--<ethereum_address>`.
+**Do not** commit your keystore to your VCS.
+
+We'll need this keystore file and the passphrase for it to
+deploy your smart contract. 
+Save the location of your keystore file
+and the passphrase you've set
+in the `.env` file we created earlier:
+
+```bash
+GATEWAY="..."
+KEYSTORE="$HOME/.ethereum/keystore/UTC--2018-12-30T12-29-11.490098600Z--<etherem_address>"
+KEYSTOREPASS="<keystore_passphrase>"
+```
+
+To deploy a contract and and make contract calls,
+we'll need our account to contain Rinkeby Ether.
+Get testnet Ether for your account by
```

Comment:
@zeddee This doesn't work unless I'm missing something. At this point the address isn't on Rinkbey so you can't get eth.

Reply:
It should work!
Don't think the ethereum networks need you to publish/broadcast an address before it becomes usable.

Did you try to do this with Twitter? I've found that (and maybe we should add this in) on Twitter,
you have to post your address as `0x<address_name>`, and then copy the permalink to that tweet and paste it into faucet.rinkeby.io

Reply:
I'm not sure about the `geth --datadir . account new` command, what network is that address on? I have been through the fawcet step before, but typically with addresses from Metamask.

Reply:
hey chris, it _should_ work. here's a video where i just ran a similar command (not exactly `geth --datadir ...`, but i promise it's the same thing) and got the faucet to give me rinkeby eth (without having to run a command with `--rinkeby`): https://drive.google.com/file/d/1kNgyf3drKgb8Llh00qlJAAFJKAczLT--/view?usp=sharing

i know this would work becauase it's something to do with how ethereum manages addresses and eth in general, and i can get on a call to have a chat about it; but tl;dr, it _should_ work

Reply:
i.e. all `geth account new` commands are net agnostic: a wallet you create there would work on mainnet or any of the testnets, i.e. you can send and receive ether on any of these nets without further configuration.

Reply:
OK, I'll take your word for it for now.

Reply:
;)